build_tools/build_libtorch.sh

```diff
@@ -102,19 +104,39 @@ fi
 
 checkout_pytorch() {
   if [[ ! -d "$PYTORCH_ROOT" ]]; then
-    git clone https://github.com/pytorch/pytorch "$PYTORCH_ROOT"
+    git clone --depth 1 --single-branch --branch "${PYTORCH_BRANCH}" https://github.com/pytorch/pytorch "$PYTORCH_ROOT"
   fi
   cd "$PYTORCH_ROOT"
-  git fetch --all
-  git checkout origin/"${PYTORCH_BRANCH}"
-  git submodule update --init --recursive
+  for dep in protobuf pocketfft cpuinfo FP16 psimd fmt sleef pybind11 onnx flatbuffers foxi; do
+    git submodule update --init --depth 1 -- third_party/$dep
+  done
 }
 
 build_pytorch() {
   cd "$PYTORCH_ROOT"
   # Uncomment the next line if you want to iterate on source builds
-  python setup.py clean
-
+  # python setup.py clean
+
+  sed -i.bak -E 's/^[[:space:]]+check_submodules()/#check_submodules()/g' setup.py
+  # remove after https://github.com/pytorch/pytorch/issues/81178 is fixed
+  sed -i.bak -E 's/namespace ao {/namespace ao {\n#include <ATen\/native\/ao_sparse\/quantized\/cpu\/packed_params.h>/g' aten/src/ATen/native/ao_sparse/quantized/cpu/qlinear_deserialize.cpp
+  sed -i.bak -E 's/namespace ao {/namespace ao {\n#include <ATen\/native\/ao_sparse\/quantized\/cpu\/packed_params.h>/g' aten/src/ATen/native/ao_sparse/quantized/cpu/qlinear_serialize.cpp
+
+  if ! command -v ninja &> /dev/null
+  then
+    echo "You should probably install Ninja using pip install ninja"
+    USE_NINJA=0
```

Comment:
I'm not 100% sure if this works - it used to be necessary but now supposedly isn't but still appears in some places (e.g. [here](https://github.com/pytorch/pytorch/blob/c3e089a0472f8ea1d6960d35297fd9c35f260d2d/tools/setup_helpers/cmake.py#L31))

Reply:
just hard fail if ninja is not available. We require it to be installed as part of requirements.txt anyway